### PR TITLE
Update feature/deploymerge branch with latest dev

### PR DESCRIPTION
**Pull request by nzpr**
This PR merged recent dev into deploymerge feature branch. The only thing changes is that test https://github.com/rchain/rchain/pull/3422#discussion_r645309983 is not portede, as it might be not representative. We need to declare all merging cases as conflicting. This will be done in another PR.

### CODE REVIEW

File: casper/src/test/scala/coop/rchain/casper/batch1/StateMergerSpec.scala [hunk 1]
```diff
@@ -1,226 +0,0 @@
...
-    baseMergeabilityCases.map(t => evalTestCase(t._1, t._2)).map(_.runSyncUnsafe())
-  }
-
-  it should "respect peak mergeability rules" in {
-    peekMergeabilityCases.map(t => evalTestCase(t._1, t._2)).map(_.runSyncUnsafe())
-  }
-
-  // TODO remove this and enable joinMergeabilityCases
-  it should "declare conflicts when produce touch join" in {
-    touchingJoinShouldBeAlwaysConflicting.map(t => evalTestCase(t._1, t._2)).map(_.runSyncUnsafe())
```
Comment (zsluedem) [inline casper/src/test/scala/coop/rchain/casper/batch1/StateMergerSpec.scala:37]: So this one is already removed?

Comment (zsluedem) [reply casper/src/test/scala/coop/rchain/casper/batch1/StateMergerSpec.scala:37]: You can put your little test in https://github.com/rchain/rchain/pull/3422/files#diff-51408dbb416e703418d8fe04e6c3f95d8c8cf2ecd8267e526842def020b355ad

Comment (nzpr) [reply casper/src/test/scala/coop/rchain/casper/batch1/StateMergerSpec.scala:37]: @zsluedem I was thinking to add this in deploymerge branch

Comment (zsluedem) [reply casper/src/test/scala/coop/rchain/casper/batch1/StateMergerSpec.scala:37]: You mean another pr for that?

Comment (tgrospic) [reply casper/src/test/scala/coop/rchain/casper/batch1/StateMergerSpec.scala:37]: @nzpr Please update PR description with information about what you are thinking to transfer to a new PR.

**Review by tgrospic — APPROVED** [review]
LGTM

**Review by zsluedem — APPROVED** [review]
LGTM

File: casper/src/test/scala/coop/rchain/casper/batch1/StateMergerSpec.scala (hunk 1 above)
Comment (nzpr) [reply casper/src/test/scala/coop/rchain/casper/batch1/StateMergerSpec.scala:37]: I think given this <https://github.com/rchain/rchain/pull/3424/checks?check_run_id=2743790800> we might need to take a deeper look into joins. I guess makes sense to declare all joins conflicting  as @zsluedem did  and make sure test is passing or to check why they are failing. No need to add my new test, as it looks like it does not cover all cases.

**Pull request merged**